lib/logidze/versioned_association.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+module Logidze
+  module VersionedAssociation
+    def load_target
+      target = super
+
+      return target if inversed
+
+      time = owner.log_data.current_version.time
+
+      if target.is_a? Array
+        target.map! do |object|
+          object unless has_logidze? object
+
+          object.at(time)
+        end.compact!
+      elsif has_logidze? target
+        target.at!(time)
+      end
+
+      target
+    end
+
+    def has_logidze?(object)
```

Comment:
Let's add `has_logidze?` class method in `has_logidze.rb` to simplify the check.